artichoke-backend/src/extn/core/kernel/mod.rs

```diff
@@ -108,6 +120,57 @@ impl Kernel {
         }
     }
 
+    unsafe extern "C" fn integer(mrb: *mut sys::mrb_state, _slf: sys::mrb_value) -> sys::mrb_value {
+        let interp = unwrap_interpreter!(mrb);
+
+        match integer::Args::extract(&interp) {
```

Comment:
the pattern I've been using in GH-288 that I like is:

```rust
let (arg, base) = mrb_get_args!(mrb, required = 1, optional = 1);
let interp = unwrap_interpreter!(mrb);
let result = integer::Args::extract(arg, base).and_then(|args| integer::method(&interp, args));
match result {
    Ok(value) => value.inner(),
    Err(variant1) => FooError::raisef(interp, . ..),
    Err(...) => ...,
}
```

This gets rid of some of the nesting and makes the error handling cleaner since both `Args::extract` and `integer::method` share an error enum.